lib/useful_parsers.ml

```diff
@@ -0,0 +1,134 @@
+open Core
+open Parser
+open Parser.Let_syntax
+
+let year = times (Char.digit |> ignore_m) 4
+let slash = Char.one_of [ '/' ] |> ignore_m
+let dash = Char.one_of [ '-' ] |> ignore_m
+let two_digits = times (Char.digit |> ignore_m) 2
+let three_letter_month = times (Char.alpha |> ignore_m) 3
+let space = Char.space |> ignore_m
+let colon = Char.one_of[':'] |> ignore_m
+let t = Char.one_of['T'] |> ignore_m
```

Comment:
nitpick: I'd call this `capital_t` since we typically use `t` to represent a value of type `t`